src/haddock/modules/topology/topoaa/__init__.py

```diff
@@ -94,7 +94,16 @@ def get_md5(ensemble_f):
 
     def _run(self):
         """Execute module."""
-        molecules = make_molecules(self.params.pop('molecules'))
+        if self.order == 0:
+            molecules = make_molecules(self.params.pop('molecules'))
+
+        else:
+            # in case topoaa is not the first step
+            # it takes the input molecules from the previous step and
+            # when topoaa is not the first step, we only need the first model
+            # because all models will have the same identity.
+            _molecules = self.previous_io.retrieve_models()[0]
+            molecules = make_molecules([_molecules.rel_path], no_parent=True)
```

Comment:
very clear, great!

Reply:
thanks. I further improved the comment :+1: